Traffic_Lights/Traffic_Lights/Traffic_Lights/main.asm

```diff
@@ -0,0 +1,170 @@
+;
+; Traffic_Lights.asm
+;
+; Created: 28/11/2018 12:48:16
+; Author : Lukas Zagata, Daniela Porcescu
+;
+
+
+; Replace with your application code
+START:
+	
+	; Setup DDR for LEDs
+	LDI R24, (1 << PA0)
+	OUT DDRA, R24 ; set RED_CAR led to output
+
+	LDI R24 , (1 <<	PC6)
+	OUT DDRC, R24 ; set GREEN_CAR led to output
+
+	LDI R24 , (1 <<	PG2)
+	OUT DDRG, R24 ; set YELLOW_CAR led to output
+
+	LDI R24 , (1 <<	PF7)
+	OUT DDRF, R24 ; set GREEN_PEDESTRIAN led to output
+
+	LDI R24 , (1 <<	PE4)
+	OUT DDRE, R24 ; set RED_PEDESTRIAN led to output
+
+
+
+	CHECK:
+	clr r24		;clear the register r24 so we can read a value into it
+
+	in r24, PIND	;read the value into register r24
+
+	SBRS R24, 0 ; if switch input is 0, then go to red
+	JMP TRAFFIC_LIGHT_ON
+
+TRAFFIC_LIGHT_OFF:
+
+	ldi r23, 0xFF
+	out portg, r23	;Turning on YELLOW_CAR LED
+
+	ldi r22, 0x00
+	out portf, r22
+	out portc, r22	;Tunring off the rest of LEDs
+	out porte, r22
+
+	ldi  r18, 82
+    ldi  r19, 43
+    ldi  r20, 0
+L1: dec  r20
```

Comment:
You should reset the register value before entering a loop. The two inner loops will execute 255 times every time exept the first time.